cdc_vaccines/delphi_cdc_vaccines/pull.py

```diff
@@ -0,0 +1,162 @@
+
+    # Conform FIPS
+    df["fips"] = df["fips"].apply(lambda x: f"{int(x):05d}")
+    df["timestamp"] = pd.to_datetime(df["date"])
+    # Drop unnecessary columns (state is pre-encoded in fips)
+    try:
+        df.drop(drop_columns, axis=1, inplace=True)
+    except KeyError as e:
+        raise ValueError(
+            "Tried to drop non-existent columns. The dataset "
+            "schema may have changed.  Please investigate and "
+            "amend drop_columns."
+        ) from e
+    # timestamp: str -> datetime
+    df.columns = ["fips",
+                  "cumulative_counts_tot_vaccine",
+                  "cumulative_counts_tot_vaccine_12P",
+                  "cumulative_counts_tot_vaccine_18P",
+                  "cumulative_counts_tot_vaccine_65P",
+                  "cumulative_counts_part_vaccine",
+                  "cumulative_counts_part_vaccine_12P",
+                  "cumulative_counts_part_vaccine_18P",
+                  "cumulative_counts_part_vaccine_65P",
+                  "timestamp"]
+    df_dummy = df.loc[(df["fips"]!='00000') & (df["timestamp"] == min(df["timestamp"]))].copy()
+    #handle fips 00000 separately
+    df_oth = df.loc[((df["fips"]=='00000') &
+        (df["timestamp"]==min(df[df['fips'] == '00000']['timestamp'])))].copy()
+    df_dummy = pd.concat([df_dummy, df_oth])
+    df_dummy.loc[:, "timestamp"] = df_dummy.loc[:, "timestamp"] - pd.Timedelta(days=1)
+    df_dummy.loc[:, ["cumulative_counts_tot_vaccine",
+                    "cumulative_counts_tot_vaccine_12P",
+                    "cumulative_counts_tot_vaccine_18P",
+                    "cumulative_counts_tot_vaccine_65P",
+                    "cumulative_counts_part_vaccine",
+                    "cumulative_counts_part_vaccine_12P",
+                    "cumulative_counts_part_vaccine_18P",
+                    "cumulative_counts_part_vaccine_65P",
+                    ]] = 0
+
+    df =pd.concat([df_dummy, df])
+    # Obtain new_counts
+    df.sort_values(["fips", "timestamp"], inplace=True)
+    df["counts_tot_vaccine"] = df["cumulative_counts_tot_vaccine"].diff()  # 1st discrete difference
+    df["counts_tot_vaccine_12P"] = df["cumulative_counts_tot_vaccine_12P"].diff()
+    df["counts_tot_vaccine_18P"] = df["cumulative_counts_tot_vaccine_18P"].diff()
+    df["counts_tot_vaccine_65P"] = df["cumulative_counts_tot_vaccine_65P"].diff()
+    df["counts_part_vaccine"] = df["cumulative_counts_part_vaccine"].diff()
+    df["counts_part_vaccine_12P"] = df["cumulative_counts_part_vaccine_12P"].diff()
+    df["counts_part_vaccine_18P"] = df["cumulative_counts_part_vaccine_18P"].diff()
+    df["counts_part_vaccine_65P"] = df["cumulative_counts_part_vaccine_65P"].diff()
+
+    rem_list = [ x for x in list(df.columns) if x not in ['timestamp', 'fips'] ]
+    # Handle edge cases where we diffed across fips
+    mask = df["fips"] != df["fips"].shift(1)
+    df.loc[mask, rem_list] = np.nan
+    print(rem_list)
```

Comment:
was this for debugging or should it be logged?

Reply:
Removed!